src/terminal/parser/stateMachine.cpp

```diff
@@ -1821,6 +1808,118 @@ bool StateMachine::FlushToTerminal()
     return success;
 }
 
+// Disable vectorization-unfriendly warnings.
+#pragma warning(push)
+#pragma warning(disable : 26429) // Symbol '...' is never tested for nullness, it can be marked as not_null (f.23).
+#pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead (bounds.1).
+#pragma warning(disable : 26472) // Don't use a static_cast for arithmetic conversions. Use brace initialization, gsl::narrow_cast or gsl::narrow (type.1).
+
+// Returns true for C0 characters and C1 [single-character] CSI.
+constexpr bool isActionableFromGround(const wchar_t wch) noexcept
+{
+    // This is equivalent to:
+    //   return (wch <= 0x1f) || (wch >= 0x7f && wch <= 0x9f);
+    // It's written like this to get MSVC to emit optimal assembly for findActionableFromGround.
+    // It lacks the ability to turn boolean operators into binary operations and also happens
+    // to fail to optimize the printable-ASCII range check into a subtraction & comparison.
+    return (wch <= 0x1f) | (static_cast<wchar_t>(wch - 0x7f) <= 0x20);
+}
+
+[[msvc::forceinline]] static size_t findActionableFromGroundPlain(const wchar_t* beg, const wchar_t* end, const wchar_t* it) noexcept
+{
+#pragma loop(no_vector)
+    for (; it < end && !isActionableFromGround(*it); ++it)
+    {
+    }
+    return it - beg;
+}
+
+static size_t findActionableFromGround(const wchar_t* data, size_t count) noexcept
+{
+    // The following vectorized code replicates isActionableFromGround which is equivalent to:
+    //   (wch <= 0x1f) || (wch >= 0x7f && wch <= 0x9f)
+    // or rather its more machine friendly equivalent:
+    //   (wch <= 0x1f) | ((wch - 0x7f) <= 0x20)
+#if defined(TIL_SSE_INTRINSICS)
+
+    auto it = data;
+
+    for (const auto end = data + (count & ~size_t{ 7 }); it < end; it += 8)
+    {
+        const auto wch = _mm_loadu_si128(reinterpret_cast<const __m128i*>(it));

```

Comment:
Yeah exactly. There used to be performance advantage to the aligned load ops, but nowadays unaligned loads on aligned data perform exactly like aligned loads on aligned data, which is why barely anyone uses the non-u variants now. (But we have to use unaligned loads anyways because our data isn't aligned.)